python/python/tests/test_optimize.py

```diff
@@ -123,6 +126,41 @@ def check_index(has_knn_combined):
     check_index(has_knn_combined=True)
 
 
+def test_index_remapping_multiple_rewrite_tasks(tmp_path: Path):
+    base_dir = tmp_path / "dataset"
+    ds = lance.write_dataset(
+        create_table(min=0, max=1, nvec=300), base_dir, mode="overwrite"
+    )
+    ds = ds.create_index(
+        "vector",
+        index_type="IVF_PQ",
+        num_partitions=4,
+        num_sub_vectors=2,
+    )
+
+    assert ds.has_index
+
+    tbl = create_table(min=0, max=1, nvec=200)
+    ds = lance.write_dataset(tbl, base_dir, mode="append")
```

Comment:
Does it compact non-index files into one?

Shall we write new data twice to create two fragments that are not indexed, and verify that they are combined later?

Reply:
We can.  There are other tests ensuring that two fragments get combined into one (without any indexing being involved).  But it's simple enough to make this test more sophisticated.  I'll put two fragments on both sides.

Reply:
I've updated the test.